ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -417,7 +423,62 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
    */
   private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, AdminRequest adminRequest)
       throws StoreException, IOException {
-    return null;
+    boolean hostHealthy = true; //used to determine if the host is ever unhealthy
+
+    //Finds all replicas of every partition that have this server's hostName
+    List<PartitionId> partitionsWithHostName = new ArrayList<>();
+    StorageManager storageManager = (StorageManager)this.storeManager;
+    for(Map.Entry<PartitionId, DiskManager> entry : storageManager.getPartitionToDiskManager().entrySet()){
+      for(ReplicaId replica : entry.getKey().getReplicaIds()){
+        if (replica.getDataNodeId().getHostname() == currentNode.getHostname()) {
+          partitionsWithHostName.add(entry.getKey());
+          break;
+        }
+      }
```

Comment:
if the partition id is in the `storageManager`, that means this host has replicas of this partition. we don't have to do this for loop again. we can just do
```
List<PartitionId> partitionsInThisHost = new ArrayList<>(storageManager.getPartitionToDiskManager().keys());
```

Reply:
Yeah I didn't know that these partitions were local to the server but instead I thought we were a spectator to all partitions. Replaced that segment